docs/hooks/new-language.rst

```diff
@@ -1,105 +1,120 @@
 
-Language agnostic test suite
-----------------------------
+What is a hooks handler?
+------------------------
 
-Dredd hooks language abstraction bridge comes with `the language agnostic test suite <https://github.com/apiaryio/dredd-hooks-template>`__. It’s written in Gherkin - language for writing `Cucumber <https://github.com/cucumber/cucumber/wiki/A-Table-Of-Content>`__ scenarios and `Aruba CLI testing framework <https://github.com/cucumber/aruba>`__ and it tests your new language handler integration with CLI Dredd and expected behavior from user’s perspective.
+Hooks handler is a process running separately from Dredd, usually started as Dredd's child process using the :option:`--language` option. When Dredd performs testing, it communicates with the hooks handler over TCP socket. The hooks handler runs hooks for each HTTP transaction and lets Dredd know whether something got modified.
 
-What to implement
------------------
 
-If you want to write a hook handler for your language you will have to implement:
+Hooks handler life cycle
+------------------------
 
--  CLI Command runnning TCP socket server
+#. Dredd starts the command given in the :option:`--language` option as its child process (subprocess). Paths to files with hooks given in :option:`--hookfiles` are resolved to absolute paths and given to the child process as arguments.
+#. The hooks handler reads paths to hooks from arguments and loads the hooks code.
+#. The hooks handler opens a TCP socket on port 61321.
+#. Dredd waits for a moment and then tries to connect to ``http://127.0.0.1:61321``. In case the connection is not successful, it exits with status code 3.
+#. For each :ref:`type of hooks <types-of-hooks>` Dredd creates a message and sends it to the socket. The message contains UUID and serialized :ref:`transaction object <transaction>` (or an array of them, in case of `beforeAll`, `afterAll`). Individual messages are sent as UTF-8 JSON documents separated by a newline.
+#. Hooks handler reads a message, calls a corresponding hook code, and sends back a message with modified contents.
+#. Dredd awaits a message with corresponding UUID. Once it arrives, Dredd overwrites its internal HTTP transaction data with the ones from the incoming message.
 
-   -  `Must return message ``Starting`` to stdout <https://github.com/apiaryio/dredd-hooks-template/blob/master/features/tcp_server.feature#L5>`__
+.. image:: ../_static/images/hooks-handler.png
```

Comment:
Looks like this

![hooks-handler](https://user-images.githubusercontent.com/283441/60040297-edd7c300-96b8-11e9-9e0e-ab0fd47f1f2a.png)

Reply:
Nice! Perhaps it would be rendered differently in the docs, but the GitHub preview here is a little hard to read, I'd increase the font size of the vertical text (or perhaps find a way to make them horizontal).

Reply:
How do you like it rendered? Does your comment still apply? http://dredd.org/en/latest/hooks/new-language.html#hooks-handler-life-cycle

Reply:
It's readable on my MacBook Pro display, and my iPad. The vertical text is nearly unreadable on my 4K main display though.

Reply:
Bah, that's first world problems then :trollface: I might try to make it bold at least, or something like that.